### PR TITLE
Feature etb#96 broker lock

### PR DESCRIPTION
**Pull request by jrachucki-n7s**

### CODE REVIEW

File: FreeRTOS/FreeRTOSConfig.h [hunk 1]
```diff
@@ -69,8 +69,13 @@
...
 #endif
 
-__attribute__((section(".sdramMemorySection")))
-static uint8_t ucHeap[configTOTAL_HEAP_SIZE];
+/**
+ * @brief FreeRtos heap declaration
+ * First attribute puts heap into sram
+ * Second suppresses warnings
+ */
+__attribute__((section(".sdramMemorySection")))__attribute__((unused))
```
Comment (Glamhoth) [inline FreeRTOS/FreeRTOSConfig.h]: You can have them both in the one `__attribute__`

**Review by Glamhoth — CHANGES_REQUESTED** [review]

File: src/Init/Init.c [hunk 2]
```diff
@@ -249,3 +213,33 @@ Init_setup_sdram_config(Sdramc* const sdramc)
...
+    Sdramc_init(&sdramc);
+    Sdramc_startup(&sdramc);
+    Init_setup_sdram_config(&sdramc);
+    Sdramc_performInitializationSequence(&sdramc, SystemConfig_DefaultCoreClock);
+}
+
+inline static void
+Init_setup_broker_lock()
+{
+    broker_lock = xSemaphoreCreateBinary();
```
Comment (rphh) [inline src/Init/Init.c]: `xSemaphoreCreateBinaryStatic`?

File: FreeRTOS/FreeRTOSConfig.h (hunk 1 above)
Comment (jrachucki-n7s) [reply FreeRTOS/FreeRTOSConfig.h]: fixed in new commit

**Review by Glamhoth — DISMISSED** [review]

File: FreeRTOS/FreeRTOSConfig.h [hunk 3]
```diff
@@ -69,8 +69,13 @@
 #define configAPPLICATION_ALLOCATED_HEAP        1   
 #endif
 
-__attribute__((section(".sdramMemorySection")))
-static uint8_t ucHeap[configTOTAL_HEAP_SIZE];
+/**
+ * @brief FreeRtos heap declaration
+ * First attribute puts heap into sram
```
Comment (Lurkerpas) [inline FreeRTOS/FreeRTOSConfig.h]: sram?

Comment (jrachucki-n7s) [reply FreeRTOS/FreeRTOSConfig.h]: Should be SDram. Fixed in new commit

File: src/Init/Init.c (hunk 2 above)
Comment (jrachucki-n7s) [reply src/Init/Init.c]: In the new commit I have provided the functionality of static mem. allocation and fixed this problem

**Review by Lurkerpas — APPROVED** [review]

**Pull request merged**